typing/typedecl_variance.ml

```diff
@@ -119,17 +89,17 @@ let compute_variance env visited vari ty =
     | Tvar _ | Tnil | Tlink _ | Tunivar _ -> ()
     | Tpackage (_, fl) ->
         let v =
-          Variance.(if mem Pos vari || mem Neg vari then full else unknown)
+          Variance.(if mem Inv vari then full else unknown)
```

Comment:
The new check here is much stronger than before, is it not?

(I think `mem Inv v` may be equivalent to `mem Pos v && mem Neg v`, not `mem Pos v || mem Neg v`.)

Reply:
Indeed, the behavior here changes. I came to the conclusion that the previous check was potentially wrong: we allow subsumption of module types through subtyping, and as a result a variable might disappear. It is only in the `Inv` case that we can guarantee that there is no way to make it disappear. I couldn't build a counter-example, but it seems better to stay on the safe side.

Reply:
I would have preferref for the PR to focus on clarifications, with no change in behavior, and then doing the change in a second PR. (For example, if we want to know if it could break programs in the wild, we can test only the second PR and get the nice readability benefits of the first PR right now.) But now that the change is in, maybe it's okay to keep it in.